front/src/common/Map/Search/MapSearchStation.tsx

```diff
@@ -98,15 +110,27 @@ export default function MapSearchStation(props) {
     setSearchResults([...trigramResults, ...nameResults]);
   }, [trigramResults, nameResults]);
 
-  const onResultClick = (result) =>
-    onResultSearchClick({
-      result,
-      map,
-      updateExtViewport,
-      dispatch,
-      title: result.name,
-      setSearch,
-    });
+  const onResultClick = (result: SearchOperationalPointResult) => {
+    setSearch(result.name);
+
+    const coordinates = map.mapTrackSources === 'schematic' ? result.schematic : result.geographic;
+
+    const center = turfCenter(coordinates);
+
+    const newViewport = {
+      ...map.viewport,
+      longitude: center.geometry.coordinates[0],
+      latitude: center.geometry.coordinates[1],
+      zoom: 12,
+    };
+    updateExtViewport(newViewport);
+    dispatch(
+      updateMapSearchMarker({
+        title: result.name,
+        lonlat: [center.geometry.coordinates[0], center.geometry.coordinates[1]],
```

Comment:
is there more than 2 elements?

```suggestion
        lonlat: center.geometry.coordinates,
```

Reply:
Done